src/soundsourcemp3.cpp

```diff
@@ -515,6 +478,10 @@ unsigned SoundSourceMp3::read(unsigned long samples_wanted, const SAMPLE * _dest
                 break;
             }
         }
+        if (frameBefore == Stream->this_frame) {
+            // No seek, try again
+            continue;
```

Comment:
mad_frame_decode returns 0 in this case, but dos not advance to the next frame. 
If mad_frame_decode return != 0 we continue or break the loop before.